### ISSUE TITLE
Normalising quaternions

### ISSUE THREAD

**Issue by pshanthraj** [mikedwhite/lightning-vae3d#1]
https://github.com/mikedwhite/lightning-vae3d/blob/bc052cc69a2c0f6e0419831f180ab584a58f8610/lvae3d/models/ResNetVAE_3Dqu.py#L187
https://github.com/mikedwhite/lightning-vae3d/blob/bc052cc69a2c0f6e0419831f180ab584a58f8610/lvae3d/models/ResNetVAE_3Dqu.py#L188

Softmax + sqrt will restrict reconstructions to only positive (unit) quaternions. To avoid this, suggest replacing lines 187 and 188 with:

x_len = torch.sqrt(torch.sum(x*x,dim=1, keepdim=True))
x = x/torch.cat((x_len,x_len,x_len,x_len),dim=1)

Probably need to safeguard against x_len = 0. torch.expand might be a better option to torch.cat.

**Comment by mikedwhite** [comment]
This has now been implemented with `x_len = torch.where(x_len == 0.0, 1e-6, x_len)` to safeguard against dividing by 0 and `torch.Tensor.repeats` has been used in place of `torch.cat`.
